src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeService.java

```diff
@@ -163,6 +164,12 @@ public void saveEvent(UUID envelopeId, EventType eventType) {
         }
     }
 
+    @Transactional(readOnly = true)
+    public List<Envelope> getEnvelopes(LocalDate date) {
+        var envelopes = envelopeRepository.findEnvelopes(null, null, date);
+        return envelopes.isEmpty() ? emptyList() : ImmutableList.copyOf(envelopes);
```

Comment:
can you explain why this is done here but not in other methods?

Reply:
to return a immutable list to avoid affects